uaclient/cli/formatter.py

```diff
@@ -0,0 +1,38 @@
+import os
+import sys
+
+from uaclient.config import UAConfig
+
+
+# Class attributes and methods so we don't need singletons or globals for this
+class ProOutputFormatterConfig:
+    use_utf8 = True
+    use_color = True
+    show_suggestions = True
+
+    # Initializing the class after the import is useful for unit testing
+    @classmethod
+    def init(cls, cfg: UAConfig):
+        cls.use_utf8 = (
+            sys.stdout.encoding is not None
+            and "UTF-8" in sys.stdout.encoding.upper()
+        )
+
+        cls.use_color = (
+            sys.stdout.isatty()
+            and os.getenv("NO_COLOR") is None
```

Comment:
what if the user sets this to `NO_COLOR=0` ?

Reply:
From: https://no-color.org/

```
Command-line software which adds ANSI color to its output by default should check for a NO_COLOR environment variable that, when present and not an empty string (regardless of its value), prevents the addition of ANSI color.
```

Reply:
btw the same website says that `NO_COLOR` should be the last checked thing: that if configuration or CLI says show color then you show color regardless.
our CLI will have a no-color boolean option instead of a color=? option that you can set to things like 'yes, enabled, disabled, auto, blablabla' so we good CLI wise
but the configuration is not being respected: `NO_COLOR` env means no color, I'm ANDing the cases.

If you think this is a problem, we can always rename the config defaults from `cli_color = True` to `disable_color = False` and we would get a reasonable behavior.

Reply:
(by reasonable I mean that any of the triggers would disable the colors, regardless of order, and none would need precedence)